misc/Makefile

```diff
@@ -0,0 +1,131 @@
+uname_M := $(shell uname -m 2>/dev/null || echo not)
+
+ARCH ?= $(shell echo $(uname_M) | sed -e s/i.86/i386/ -e s/arm.*/arm/ )
+ifeq ($(ARCH),x86_64)
+  ifneq ($(findstring m32,$(CFLAGS)),)
+    override ARCH := i386
+  endif
+endif
+
+ifndef ($(srcdir))
+	srcdir = $(CURDIR)/..
+endif
+# set objdir to $(O) by default (if any)
+ifndef ($(objdir))
+  objdir = $(CURDIR)/..
+endif
+
+ifneq ($(wildcard $(objdir)/.config),)
+  include $(objdir)/.config
+endif
+
+RM = rm -f
+
+COMMON_CFLAGS := -D_GNU_SOURCE $(CFLAGS) $(CPPFLAGS)
+COMMON_CFLAGS += -iquote $(srcdir) -iquote $(objdir) -iquote $(srcdir)/arch/$(ARCH)
+COMMON_LDFLAGS := -lrt -ldl -pthread $(LDFLAGS)
+ifneq ($(elfdir),)
+  COMMON_CFLAGS  += -I$(elfdir)/include
+  COMMON_LDFLAGS += -L$(elfdir)/lib
+endif
+
+COMMON_CFLAGS += -W -Wall -Wno-unused-parameter -Wno-missing-field-initializers
+
+#
+# Note that the plain CFLAGS and LDFLAGS can be changed
+# by config/Makefile later but *_*FLAGS can not.
+#
+DEMANGLER_CFLAGS   = $(COMMON_CFLAGS) $(CFLAGS_$@) $(CFLAGS_demangler)
+SYMBOLS_CFLAGS     = $(COMMON_CFLAGS) $(CFLAGS_$@) $(CFLAGS_symbols)
+LIB_CFLAGS         = $(COMMON_CFLAGS) $(CFLAGS_$@) $(CFLAGS_lib)
+LIB_CFLAGS        += -fPIC -fvisibility=hidden -fno-omit-frame-pointer
+
+DEMANGLER_LDFLAGS  = $(COMMON_LDFLAGS) $(LDFLAGS_$@) $(LDFLAGS_demangler)
+SYMBOLS_LDFLAGS    = $(COMMON_LDFLAGS) $(LDFLAGS_$@) $(LDFLAGS_symbols)
+LIB_LDFLAGS        = $(COMMON_LDFLAGS) $(LDFLAGS_$@) $(LDFLAGS_lib) -Wl,--no-undefined
+
+ifeq ($(DEBUG), 1)
+  COMMON_CFLAGS += -O0 -g
+else
+  COMMON_CFLAGS += -O2 -g
+endif
+
+ifeq ($(TRACE), 1)
+  DEMANGLER_CFLAGS  += -pg -fno-omit-frame-pointer
+  SYMBOLS_CFLAGS    += -pg -fno-omit-frame-pointer
+endif
+
+ifeq ($(COVERAGE), 1)
+  COMMON_CFLAGS += -O0 -g --coverage -U_FORTIFY_SOURCE
+  LIB_CFLAGS    += -O0 -g --coverage -U_FORTIFY_SOURCE
+  LIB_LDFLAGS   += --coverage
+endif
+
+VERSION_GIT := $(shell git describe --tags 2> /dev/null || echo v$(VERSION))
+
+all:
+
+ifneq ($(wildcard $(srcdir)/check-deps/check-tstamp),)
+	@$(MAKE) -C $(srcdir)/check-deps 
+  include $(srcdir)/check-deps/Makefile.check
+endif
+
+include $(srcdir)/Makefile.include
+
+_TARGETS += misc/demangler misc/symbols
+TARGETS  := $(patsubst %,$(objdir)/%,$(_TARGETS))
+
+DEMANGLER_SRCS := $(srcdir)/misc/demangler.c
+DEMANGLER_OBJS := $(objdir)/misc/demangler.o
+DEMANGLER_DEPS := $(objdir)/misc/demangle.o $(objdir)/misc/debug.o
+
+SYMBOLS_SRCS := $(srcdir)/misc/symbols.c
+SYMBOLS_OBJS := $(srcdir)/misc/symbols.o
+
+MISC_DEP_SRCS := $(srcdir)/utils/session.c
+MISC_DEP_SRCS += $(srcdir)/utils/demangle.c $(srcdir)/utils/rbtree.c
+MISC_DEP_SRCS += $(srcdir)/utils/utils.c $(srcdir)/utils/debug.c
+MISC_DEP_SRCS += $(srcdir)/utils/filter.c $(srcdir)/utils/dwarf.c
+MISC_DEP_SRCS += $(srcdir)/utils/auto-args.c
+MISC_DEP_SRCS += $(wildcard $(srcdir)/utils/symbol*.c)
+MISC_DEP_OBJS := $(patsubst $(srcdir)/utils/%.c,$(objdir)/misc/%.o,$(MISC_DEP_SRCS))
```

Comment:
Why did you separate it?  Isn't it only used by the `symbols`?

Reply:
@namhyung I thought these objects would be used by other misc programs.

Reply:
You can separate it when it's actually used.